core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -721,33 +719,30 @@ private suspend fun runVideoRecording(
         ).use { recording ->
 
             fun TransientSessionSettings.isFlashModeOn() = flashMode == FlashMode.ON
-            val isFrontCameraSelector =
-                camera.cameraInfo.cameraSelector == CameraSelector.DEFAULT_FRONT_CAMERA
-
-            if (currentSettings.isFlashModeOn()) {
-                if (!isFrontCameraSelector) {
-                    camera.cameraControl.enableTorch(true).await()
-                } else {
-                    Log.d(TAG, "Unable to enable torch for front camera.")
+            val isFrontFacing = camera.cameraInfo.appLensFacing == LensFacing.FRONT
+
+            var torchOn = false
+            fun setTorchOn(newTorchOn: Boolean) {
```

Comment:
We shouldn't use `CameraState` here since `CameraState` is supposed to be showing the _actual_ state of the camera, which can update asynchronously from the _expected_ state of the camera (the `torchOn` var). If we used `CameraState`, it's possible that once we've told the torch to turn on, the recording could end before the torch actually turns on, and then we would skip turning off the torch. This would cause the torch to remain on when it shouldn't.